util/util.go

```diff
@@ -491,6 +491,21 @@ func (o MachinesByCreationTimestamp) Less(i, j int) bool {
 	return o[i].CreationTimestamp.Before(&o[j].CreationTimestamp)
 }
 
+// MachinesByVersion sorts the list of Machine by spec.version, using their names as tie breaker.
+// machines with no version are placed lower in the order.
+type MachinesByVersion []*clusterv1.Machine
```

Comment:
I would keep this in util collection as a private object, so we don't expose it
otherwise tests are required

Reply:
Keeping it here along with other similar types like `MachinesByCreationTimestamp `. Added tests.